src/sql/base/browser/ui/table/tableDataView.ts

```diff
@@ -255,3 +255,41 @@ export class TableDataView<T extends Slick.SlickData> implements IDisposableData
 		this._findArray = [];
 	}
 }
+
+export class SlickTableDataView<T extends Slick.SlickData> extends Slick.Data.DataView<T>
+	implements IDisposableDataProvider<T>
+{
+	push(items: Array<T>): void;
+	push(item: T): void;
+	push(input: T | Array<T>): void {
+		let inputArray = new Array();
+		if (Array.isArray(input)) {
+			inputArray.push(...input);
+		} else {
+			inputArray.push(input);
+		}
+		if (inputArray.length > 0 && inputArray[0]['id'] === undefined) {
+			inputArray.forEach(item => item.id = generateUuid());
+		}
+
+		this.beginUpdate();
+		inputArray.forEach(item => this.addItem(item));
+		this.endUpdate();
+	}
+
+	public setData(data: T[]): void {
```

Comment:
Hmm. Not sure if I'm a fan of changing the actual data objects...

Reply:
I need id property for rowDetails plugin. Maybe this is not the best place to do this. I will create it during the construction of data source object in the transformData method of table component.